### ISSUE TITLE
[FEATURE] Compact I/O list for operations in manager

### ISSUE THREAD

**Issue by dvnstrcklnd** [aquariumbio/aquarium#487]
When planning operations for the Duke COVID testing workflow, we need to be able to plan operations with 96 input samples each, and batch 4 of these operations together based on the plan or output sample. The list of 96 input samples is mostly uninformative, and necessitates a lot of scrolling to find the 4 operations to be batched:

![image](https://user-images.githubusercontent.com/7586251/101562255-0fa03500-3995-11eb-910e-bba53dba2b29.png)

I would like to be able to select all 4 operations without scrolling. This will streamline the batching of operations and make the process less error-prone. 

This is a fairly high priority because we anticipate running this workflow at scale early in the new year.

**Comment by spcoon** [comment]
for this exact use case—where you want to select all four operations without scrolling—the "all" button at the top of the interface selects all four operations. clicking "schedule" batches them into a job.

with technician handoff, the all button is replaces by a checkbox that selects all operations.

**Comment by dvnstrcklnd** [comment]
Clarification: I mean all 4 operations _that I intend to batch_, not all operations in the list.

**Comment by spcoon** [comment]
ok, so not this particular case since there are only four operations.

i need you to describe what's necessary operation information and what's not. there's not much design thinking to do here aside from simplifying the display. you said "mostly uninformative" above. what parts are informative?

**Comment by bjkeller** [comment]
It seems like it might be a good idea to hide a long I/O list by default.

**Comment by spcoon** [comment]
i agree, ben. what form do i need to fill out to specify a solution of "if more than five lines, display **more** link that expands on click"

?

**Comment by dvnstrcklnd** [comment]
The plan and the output sample are what's informative. The intent is that they will be batched on the output sample name, which I intend to have a 1:1 relationship with the plan.

I am agnostic as to whether we implement a filter function or collapse the list by default. The latter seems like it would be easier, but probably not as generally usable and thus less desirable in the long-term.

**Comment by spcoon** [comment]
ok, i re-read the original post. this is a bit trickier than what i originally thought. i'm going to prioritize this for monday.

**Comment by spcoon** [comment]
@dvnstrcklnd i suggest we just add sorting to this interface and truncate operations that generate > five lines. not sure what other quick fixes would make sense.

**Comment by bjkeller** [comment]
This is v2-tech-handoff, correct?

**Comment by marikoja** [comment]
@dvnstrcklnd @bjkeller @spcoon this is the simplest solution I could come up with that I think solves for Duke's need. 

Inputs start hidden:
![image](https://user-images.githubusercontent.com/31191468/105103541-9d2b7600-5a65-11eb-80b5-dc148b08a64c.png)

Toggle switch shows inputs: 
![image](https://user-images.githubusercontent.com/31191468/105103589-b03e4600-5a65-11eb-82e0-a697e5fd592c.png)

**Comment by dvnstrcklnd** [comment]
I think this checks the box for the Duke scenario, although it might not be the most future-proof way to do it.

**Comment by marikoja** [comment]
@dvnstrcklnd I agree with you about that this solution may not be future proof. However this code is in angular and will be re-written into react I feel confident this is a solution that minimizes additional tech-debt.
